Add dark mode for docs

FestiveAkp opened the pull request:
I thought I'd take a stab at creating a dark mode version of the docs:

<img width="947" alt="laravel-2" src="https://user-images.githubusercontent.com/20550003/126120092-08415036-e7ae-4225-b69f-233509990ee4.png">

<img width="1367" alt="laravel-9" src="https://user-images.githubusercontent.com/20550003/126120757-acd05d71-cfe3-4136-8dd3-f51d350ea078.png">

Updated syntax highlighting under dark mode:

![laravel-1](https://user-images.githubusercontent.com/20550003/126120102-573c6eb7-9b92-48a4-8537-d2d32e7c5cc2.png)

<img width="947" alt="laravel-6" src="https://user-images.githubusercontent.com/20550003/126120809-f700fae6-217b-4da1-8d5d-5420ea34af1b.png">

Updated footer:

<img width="1229" alt="laravel-8" src="https://user-images.githubusercontent.com/20550003/126120750-b5ded917-8c38-4d18-94ff-f916b8d7d6c8.png">


The initial color setting is based on `prefers-color-scheme`. You can toggle between light and dark with the button on the header:

<img width="219" alt="laravel-5" src="https://user-images.githubusercontent.com/20550003/126120446-612c18d2-c1c6-41b8-90a1-041f62c72d9a.png"> <img width="217" alt="laravel-4" src="https://user-images.githubusercontent.com/20550003/126120452-4350fd3c-c381-4027-9b37-ff0071b43f17.png">

The mobile version:

<img width="305" alt="laravel-11" src="https://user-images.githubusercontent.com/20550003/126120921-113591f9-5be9-4113-8e17-85900effa546.png"> <img width="305" alt="laravel-10" src="https://user-images.githubusercontent.com/20550003/126123191-1917a79c-097f-4917-ba2e-ddd9b879e5fc.png">

<img width="305" alt="laravel-12" src="https://user-images.githubusercontent.com/20550003/126120932-3d0d02dd-8dfb-4f14-885d-b3326bd7c5b1.png"> <img width="306" alt="laravel-14" src="https://user-images.githubusercontent.com/20550003/126121590-3417289f-9935-4ef1-9e99-a13d0bdd85a2.png">

 For the time being, dark mode has only been implemented on pages under /docs/*, the rest of the website should still be unchanged and only have light mode. This is my first open-source contribution, so any and all feedback would be definitely appreciated.

taylorotwell commented:
When I click a link in the documentation to go to another page it switches back to light mode?

FestiveAkp commented:
That should fix it, I wasn't saving the chosen theme anywhere. Now it's saved in localStorage and gets fetched on every page load.

taylorotwell commented:
So, once it's in local storage - it is then impossible to let the OS dictate the color? I can't seem to get OS to dictate color once I manually select a color.

JonoB commented:
> So, once it's in local storage - it is then impossible to let the OS dictate the color? I can't seem to get OS to dictate color once I manually select a color.

Many website offer 3 modes:
- OS (default)
- light
- dark

NicolasPereira commented:
Fantastic idea! 
I already use dark mode on most sites I use, like github, twitter etc.

I believe it is a valid implementation.

FestiveAkp commented:
> So, once it's in local storage - it is then impossible to let the OS dictate the color? I can't seem to get OS to dictate color once I manually select a color.

Yes, at the moment it toggles between light and dark mode and keeps it's choice regardless of the system setting (For example, I believe https://chakra-ui.com/ does it like this). I could add a third state to the button that respects the OS preference, or change it to a dropdown that has choices for the three modes. I like how clean the single button looks so I'd try that first, but I've seen it done lots of different ways, so let me know what you think.

corbosman commented:
I would really like it to respect the OS setting. The less you have to think about dark/light mode the better.

lhsazevedo commented:
This is awesome!

One thing that seems a off to me is the footer background color. I think it is too dark, the contrast with the page background is not so subtle as in the light theme.

juancho48 commented:
I have a PCT cataract in my right eye, makes reading the light docs a little hard for me some days, Thanks for this PR!

taylormaguire commented:
Very supportive of this. Proposed screenshots looks good. 

I would say it's important to allow for 3 modes though
OS Managed
Light (forced)
Dark (forced)

Optional Mode idea
Paper Mode (a softer yellow/book color mode, that is seen on some ebook readers)

there are a number of sites that support this.

negoziator commented:
Here's my feedback.

Love it.

Period. ❤️

AbdianRizky commented:
Congrats on your first open source contribution, it will make a huge impact to the community if merged this PR.

Maybe it will be great if we can customize our personal color mode.

Cheers.😄

genericmilk commented:
Bloomin love this, top work guys! To echo others, OS sensitive mode would be a fab idea!

HDVinnie commented:
Looks great!

Nafario commented:
Looks good...

mnightingale commented:
> So, once it's in local storage - it is then impossible to let the OS dictate the color? I can't seem to get OS to dictate color once I manually select a color.

Perhaps the implementation should only save the preference to localStorage if it differs from the system default?
i.e. use `window.matchMedia('(prefers-color-scheme: dark)')` to determine if the users preference needs to be stored
Light + Light => clear localstorage
Light + Dark = store dark

SBTHDR commented:
It would be nice!

reppair commented:
I love it. I e actually check for this option periodically.

About theme switching, I like the proposal from @mnightingale - use the switch as overwrite of OS setting.

Looking good! 🕶️

ConsoleTVs commented:
I would add some shadow to the panel here, the content is intended to be "floating above". Some shadow as the light version should help aliviate it. Otherwise, dark themes use lighter color variants to simulate the light above (the closest, the lighter). So I would suggest to either:

- (a) Try adding a shadow 
- (b) try a lighter variant of that gray for the panel.

![Screenshot 2021-07-19 at 22 24 01](https://user-images.githubusercontent.com/6124435/126222458-d61cc32a-87fc-47ac-a157-87105bc65062.png)

Regarding the persistance and overall, the chosing of the theme (light or dark), here's my two cents:

Major OSs support theming already, and those are available to both, CSS and JS. There's two main ways to overcome this problem. I would suggest reading: https://tailwindcss.com/docs/dark-mode

Overall, the strategy you chose depends on the goal, so:

- If you want to always respect the OS and don't allow switching (eg. use light theme when OS is dark theme): Use the CSS selector (https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme)
- If you want to allow switching + respecting the OS decision as the default, use the `window.matchMedia('(prefers-color-scheme: dark)')` as suggested by @mnightingale (https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia). The link above, to the tailwind documentation also provides an example implementation to look after.

datlechin commented:
awesome!!!!!!

FestiveAkp commented:
Thank you everyone for your suggestions! I just implemented the third color mode, which inherits the system preference. It's represented as the third state of the toggle button, a half shaded-in circle: 

<img width="639" alt="laravel-16" src="https://user-images.githubusercontent.com/20550003/126248928-e8803708-54cd-4c12-97b7-0d681f8e679a.png">

<img width="601" alt="laravel-17" src="https://user-images.githubusercontent.com/20550003/126248930-399b75d8-8d74-4c5e-bfc1-9470e6503ffb.png">

<img width="305" alt="laravel-18" src="https://user-images.githubusercontent.com/20550003/126249494-a26138b2-7cef-4a5b-89f1-7f5995f6aaa7.png">


On first visit, the site theme is set to the system preference. Then you can click the button to switch to dark mode, click it again to switch to light mode, and click a final time to go back to the system preference. Your theme preference is stored in `localStorage` (either 'system', 'light', or 'dark') and persists across page loads. You can hover over the button to see what the next color mode is. I also added an event listener to `window.matchMedia('(prefers-color-scheme: dark)')` which means that the site will update in real time if you change your OS theme!

Thank you @mnightingale for the footnotes about the build files and the package-lock.json, those have been corrected. I also moved the theme initializing code out of `<body>` and in to `<head>` to avoid FOUC. Sometime tomorrow I'll take a look at the colors for the footer as well as any other changes.

taylorotwell commented:
Don't implement changes just because someone in the thread mentions something - I will suggest any changes I want to see. No other suggested changes should be applied. 😄

reppair commented:
:rocket:

Unlimited23 commented:
This should've been done a long time ago! Thank you bro! :)

veekthoven commented:
Awesome!

gjzim commented:
Thanks for this.

P9Q commented:
Thank you for your great Job!

P9Q commented:
Thanks for your great job! 👍 

It would be great if you also customize and make the scrollbar dark (something like YouTube's scrollbar)

![Untitled-1](https://user-images.githubusercontent.com/20019720/126425910-f6b8a5fd-0bf5-4957-bf16-bfd55bea7a23.jpg)

mnaderian commented:
Finally!!

bilogic commented:
@FestiveAkp Cool!! How about the API docs as well? 🤣 https://laravel.com/api/master/index.html
